### PR TITLE
Translate `Ds\Deque`

### PR DESCRIPTION
**Pull request by Fan2Shrek**

### CODE REVIEW

File: reference/ds/ds/deque/toarray.xml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  <methodsynopsis>
+   <modifier>public</modifier> <type>array</type><methodname>Ds\Deque::toArray</methodname>
+   <void />
+  </methodsynopsis>
+  <para>
+    Convertit le deque en un &array;.
+  </para>
+  <note>
+    <para>
+        La conversion en un &array; n'est pas encore supportée.
```
Comment (Fan2Shrek) [inline reference/ds/ds/deque/toarray.xml:23]: I don't know to make this part accurate

Comment (devnexen) [reply reference/ds/ds/deque/toarray.xml:23]: this expression is used somewhere else, it s fine even though not looking "ideal".

File: reference/ds/ds/deque/allocate.xml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+ <refsect1 role="description">
+  &reftitle.description;
+  <methodsynopsis>
+   <modifier>public</modifier> <type>void</type><methodname>Ds\Deque::allocate</methodname>
+   <methodparam><type>int</type><parameter>capacity</parameter></methodparam>
+  </methodsynopsis>
+  <para>
+    Assure qu'assez de mémoire est allouée pour une capacité requise.
+    Cela élimine le besoin de réallouer l'interne à mesure que les valeurs sont ajoutées.
```
Comment (devnexen) [inline reference/ds/ds/deque/allocate.xml:18]: I would say `réallouer la mémoire interne`

File: reference/ds/ds/deque/clear.xml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+ </refnamediv>
+
+ <refsect1 role="description">
+  &reftitle.description;
+  <methodsynopsis>
+   <modifier>public</modifier> <type>void</type><methodname>Ds\Deque::clear</methodname>
+   <void />
+  </methodsynopsis>
+  <para>
+    Enlève toutes les valeurs du deque.
```
Comment (devnexen) [inline reference/ds/ds/deque/clear.xml:17]: I would say `Supprime` instead of `Enlève` in the present instances wdyt ?

**Review by devnexen — APPROVED** [review]

**Pull request merged**